### DISCUSSION TITLE
GitHub search a path in a repository via the GitHub CLI

### DISCUSSION THREAD

**Discussion by DeepspaceFreelancer** [community/community#68413] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I have this search via the code browser:

* [repo:RapidCircle/vscode path:.vscode/][1]

It searches for a path inside a repository and it does returns results.

No matter how I try it to translate it to GitHub CLI I can not succeed. The following one (Windows):

```
gh api --method=GET "search/code?q=repo:RapidCircle/vscode+path:.vscode/"
```

returns empty.

So does the following one (Ubuntu) too:

```
gh api -X GET search/code -f 'q=repo:RapidCircle/vscode+path:.vscode/'
```

Did some extra investigation and these queries do not return result either:

```
gh api -X GET search/code -f q='repo:RapidCircle/vscode' -F per_page=100
```
```
gh api --method=GET "search/code?q=repo:RapidCircle/vscode"
```

But the repo ([RapidCircle/vscode][2]) seems there. Ok. Fine. It might not be indexed ([Search for the repo in an other way][3]).

Let us look at another repo. This Github CLI does returns values:

```
gh api -X GET search/code -f q='repo:cli/cli' -F per_page=100
```

If I try to add a path variable ([works via web][4]) but not with GitHub CLI:

```
gh api -X GET search/code -f q='repo:cli/cli+path:/^internal\//' -F per_page=100
```

Same but Windows:

```
gh api --method=GET "search/code?q=repo:cli/cli+path:/^internal\//"
```

What is wrong?

A more general question would be: 

* Is there a tool out there that could translate GitHub Web searches to CLI commands?

I am guessing there is limitation on what you can do with GitHub CLI API vs real WebSearch (on the site).

  [1]: https://github.com/search?q=repo%3ARapidCircle%2Fvscode%20path%3A.vscode%2F&type=code
  [2]: https://github.com/RapidCircle/vscode
  [3]: https://github.com/search?q=RapidCircle%2Fvscode&type=repositories
  [4]: https://github.com/search?q=repo%3Acli%2Fcli%20path%3A%2F%5Einternal%5C%2F%2F%20&type=code

**Comment by DeepspaceFreelancer** [comment]
I have this question on StackOverflow too: [GitHub search a path in a repository via the GitHub CLI](https://stackoverflow.com/questions/77183019/github-search-a-path-in-a-repository-via-the-github-cli)

**Comment by look** [comment, marked as answer]
Hi @DeepspaceFreelancer, I looked into this and the issue is that the repository you're searching is a fork, and the current version of the GitHub code search API suppresses forks by default. You can search it by adding `fork:true`:

```
gh api --method=GET "search/code?q=repo:RapidCircle/vscode+path:.vscode/+fork:true"

```

**Reply by DeepspaceFreelancer** [reply]
Thank you dear Luke for giving the answer. If you want you can give the answer here in StackOverflow too and I will accept and vote it up too: [GitHub search a path in a repository via the GitHub CLI](https://stackoverflow.com/questions/77183019/github-search-a-path-in-a-repository-via-the-github-cli)

**Comment by DeepspaceFreelancer** [comment]
Yes I can confirm that the search queries with the "fork:true" work.

This one under Windows (specified by Luke @look):

```
gh api --method=GET "search/code?q=repo:RapidCircle/vscode+path:.vscode/+fork:true"
```

And here is a syntax that worked for me in Linux (Ubuntu):

```
gh api -X GET search/code -f 'q=repo:RapidCircle/vscode path:.vscode/ fork:true'
```

Please pay attention that instead of "+" there is space there.

As for these commands:

```
gh api -X GET search/code -f q='repo:cli/cli+path:/^internal\//' -F per_page=100
gh api -X GET search/code -f q='repo:cli/cli path:/^internal\//' -F per_page=100
```

Wiindows:
```
gh api --method=GET "search/code?q=repo:cli/cli+path:/^internal\//"
```

They are not working because this search pattern: /^internal\// is Regexp and it seams GitHub CLI does not know that one yet.

Commands like these (regexp free) work:

Ubuntu shell
```
gh api -X GET search/code -f q='repo:cli/cli path:internal/' -F per_page=100
```

Windows shell
```
gh api --method=GET "search/code?q=repo:cli/cli+path:internal/"
```
